crates/da-clients/celestia/src/lib.rs

```diff
@@ -0,0 +1,190 @@
+    async fn test_celestia_publish_state_diff_and_verify_inclusion() {
+        let config = CelestiaDaConfig {
+            http_provider: DEFAULT_CELESTIA_NODE.to_string(),
+            auth_token: None,
+            nid: DEFAULT_NID.to_string(),
+        };
+        // Instantiate CelestiaDaClient
+        let celestia_da_client = CelestiaDaClient::try_from(config).unwrap();
+
+        let s = "Hello World!";
+        let bytes: Vec<u8> = s.bytes().collect();
+        let state_diff = vec![bytes];
+
+        let to: [u8; 32] = [
+            0x01, 0x23, 0x45, 0x67, 0x89, 0xab, 0xcd, 0xef, 0xfe, 0xdc, 0xba, 0x98, 0x76, 0x54, 0x32, 0x10, 0x00, 0x11,
+            0x22, 0x33, 0x44, 0x55, 0x66, 0x77, 0x88, 0x99, 0xaa, 0xbb, 0xcc, 0xdd, 0xee, 0xff,
+        ];
+
+        let height_response = celestia_da_client.publish_state_diff(state_diff, &to).await;
+
+        let height_id = match height_response {
+            Ok(variable) => variable,
+            Err(error) => panic!("Problem reading: {error:?}"),
+        };
+
+        let inclusion_response = celestia_da_client.verify_inclusion(&height_id).await;
+
+        let inclusion = match inclusion_response {
+            Ok(variable) => variable,
+            Err(error) => panic!("Problem reading: {error:?}"),
+        };
+
+        assert_eq!(inclusion, DaVerificationStatus::Verified);
+
+        match inclusion {
+            DaVerificationStatus::Pending => println!("Verification Status is Pending"),
+            DaVerificationStatus::Verified => println!("Verification Status is Verified"),
+            DaVerificationStatus::Rejected(msg) => println!("Verification Status is Rejected: {}", msg),
+        }
+    }
+
+    #[tokio::test]
+    #[ignore = "Can't run without manual intervention, setup celestia-node."]
+    async fn test_max_blob_per_txn() {
+        let expected_value: u64 = 1;
+
+        let config = CelestiaDaConfig {
+            http_provider: DEFAULT_CELESTIA_NODE.to_string(),
+            auth_token: None,
+            nid: DEFAULT_NID.to_string(),
+        };
+        // Instantiate CelestiaDaClient
+        let celestia_da_client = CelestiaDaClient::try_from(config).unwrap();
+
+        let max_blobs_per_txn = celestia_da_client.max_blob_per_txn().await;
+        assert_eq!(max_blobs_per_txn, expected_value);
+    }
+
+    #[tokio::test]
+    #[ignore = "Can't run without manual intervention, setup celestia-node."]
+    async fn test_max_bytes_per_blob() {
+        let expected_value: u64 = 1974272;
+
+        let config = CelestiaDaConfig {
+            http_provider: DEFAULT_CELESTIA_NODE.to_string(),
+            auth_token: None,
+            nid: DEFAULT_NID.to_string(),
+        };
+        // Instantiate CelestiaDaClient
+        let celestia_da_client = CelestiaDaClient::try_from(config).unwrap();
+
+        let max_bytes_per_blob = celestia_da_client.max_bytes_per_blob().await;
+        assert_eq!(max_bytes_per_blob, expected_value);
+    }
+}
```

Comment:
Hmm, this looks a little too trivial to write a test case for, what do you think?

Reply:
Valid, trivial test cases.
My thinking process was to test every detail possible, hence these exist, totally a subjective call here.